src/gradle.ts

```diff
@@ -43,11 +43,20 @@ export function getTaskToPublish(
     } else {
       let task = "";
       child.stdout.pipe(split()).on("data", (line: string) => {
-        if (line.startsWith("publish -")) {
+        if (line.startsWith("artifactoryDeploy -")) {
           if (task !== "") {
```

Comment:
Are we sure that `publish` always come earlier than `artifactoryDeploy`? If not, it is necessary to add if-block here.

```js
        if (line.startsWith("artifactoryDeploy -")) {
          if (task !== "" && task !== "publish") {
            reject(new Error("Found multiple tasks to publish"));
          }
          if (task === "publish") {
            // TODO: log to say 'we found both artifactory and maven-publish so we use artifactory for release'
          }
          task = "artifactoryDeploy";
```